client/chain-spec/src/lib.rs

```diff
@@ -158,3 +158,9 @@ pub trait ChainSpec: BuildStorage + Send {
 	/// This will be used as storage at genesis.
 	fn set_storage(&mut self, storage: Storage);
 }
+
+impl std::fmt::Debug for dyn ChainSpec {
```

Comment:
Whoa, didn't know you could impl for `dyn` things! What's the difference between doing this and impling for `T where T: ChainSpec`?

Reply:
:thinking: I have no idea. Maybe someone else know

Reply:
The compiler will refuse to compile `impl<T: ChainSpec> Debug for T {}`

That being said, I don't think it's a good idea to implement `Debug` on trait objects.